modules/flowable-inmemory-data/src/test/java/org/flowable/engine/data/inmemory/activity/MemoryActivityDataManagerTest.java

```diff
@@ -0,0 +1,118 @@
+
+    @Test
+    public void testFindActivity() throws InterruptedException {
+        Deployment deployment = processEngine.getRepositoryService().createDeployment()
+                        .addClasspathResource("org/flowable/engine/test/inmemory/triggerableExecution.bpmn20.xml").deploy();
+
+        MemoryActivityInstanceDataManager actManager = getActivityInstanceDataManager();
+        try {
+            ProcessInstance instance = processEngine.getRuntimeService().startProcessInstanceByKey("triggerableExecution");
+            ActivityInstanceEntity activity = waitForActivity(instance.getProcessInstanceId(), "service1");
+            assertThat(activity).isNotNull();
+
+            assertThat(actManager.findById(activity.getId())).isNotNull();
+            assertThat(actManager.findActivityInstancesByExecutionIdAndActivityId(activity.getExecutionId(), "service1").size()).isEqualTo(1);
+            assertThat(actManager.findActivityInstancesByExecutionIdAndActivityId(activity.getExecutionId(), "foo").size()).isEqualTo(0);
+            assertThat(actManager.findUnfinishedActivityInstancesByExecutionAndActivityId(activity.getExecutionId(), "service1").size()).isEqualTo(1);
+            assertThat(actManager.findUnfinishedActivityInstancesByExecutionAndActivityId(activity.getExecutionId(), "foo").size()).isEqualTo(0);
+        } finally {
+            processEngine.getRepositoryService().deleteDeployment(deployment.getId(), true);
+            processEngine.close();
+        }
+    }
+
+    @Test
+    public void testFindActivityAsync() throws InterruptedException {
+        Deployment deployment = processEngine.getRepositoryService().createDeployment()
+                        .addClasspathResource("org/flowable/engine/test/inmemory/triggerableAsyncExecution.bpmn20.xml").deploy();
+        processEngine.getProcessEngineConfiguration().getAsyncExecutor().start();
+
+        MemoryActivityInstanceDataManager actManager = getActivityInstanceDataManager();
+        try {
+            ProcessInstance instance = processEngine.getRuntimeService().startProcessInstanceByKey("triggerableAsyncExecution");
+            ActivityInstanceEntity activity = waitForActivity(instance.getProcessInstanceId(), "service1");
+            assertThat(activity).isNotNull();
+
+            assertThat(actManager.findById(activity.getId())).isNotNull();
+            assertThat(actManager.findActivityInstancesByExecutionIdAndActivityId(activity.getExecutionId(), "service1").size()).isEqualTo(1);
+            assertThat(actManager.findActivityInstancesByExecutionIdAndActivityId(activity.getExecutionId(), "foo").size()).isEqualTo(0);
+            assertThat(actManager.findUnfinishedActivityInstancesByExecutionAndActivityId(activity.getExecutionId(), "service1").size()).isEqualTo(1);
+            assertThat(actManager.findUnfinishedActivityInstancesByExecutionAndActivityId(activity.getExecutionId(), "foo").size()).isEqualTo(0);
```

Comment:
lines 68-72 could be replaced with:
<pre>
            assertThat(actManager.findActivityInstancesByExecutionIdAndActivityId(activity.getExecutionId(), "service1")).hasSize(1);
            assertThat(actManager.findActivityInstancesByExecutionIdAndActivityId(activity.getExecutionId(), "foo").isEmpty());
            assertThat(actManager.findUnfinishedActivityInstancesByExecutionAndActivityId(activity.getExecutionId(), "service1")).hasSize(1);
            assertThat(actManager.findUnfinishedActivityInstancesByExecutionAndActivityId(activity.getExecutionId(), "foo").isEmpty());
</pre>

Reply:
There are other occurrences of this same change in other tests that I will not enumerate.

Reply:
updated this and all similar occurrences to use more appropriate assertion methods.